### DISCUSSION TITLE
Compatibility with Safari 12?

### DISCUSSION THREAD

**Discussion by simonhrogers** [muxinc/media-chrome#736] [Q&A] [answered]
Does this plugin support Safari 12? 

I’m using `media-chrome/react` with Next.js, which still purports to support Safari 12 (I wish the truth were quite so straightforward, but it does with some painful tinkering). 

In Safari I get the following errors when attempted to load these components:

`W.customElements.upgrade is not a function. (In 'W.customElements.upgrade(e)', 'W.customElements.upgrade' is undefined)`

What measures can I take to avoid this?

Thanks!

**Comment by luwes** [comment, marked as answer]
good question! the Safari browser is a tricky one, it didn't support certain required features until fairly late.
The latest Safari version is 16 and we support until version 14 I believe. 

Beyond that the market share really drops off so in most cases it's not worth keeping up support for it.
https://www.stetic.com/market-share/browser/

Hope this helps.
